src/sentry/static/sentry/app/views/organizationEventsV2/modalPagination.jsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'react-emotion';
+import isPropValid from '@emotion/is-prop-valid';
+import {omit} from 'lodash';
+
+import {t} from 'app/locale';
+import Link from 'app/components/links/link';
+import SentryTypes from 'app/sentryTypes';
+import InlineSvg from 'app/components/inlineSvg';
+import space from 'app/styles/space';
+
+const ModalPagination = props => {
+  const {location, event} = props;
+
+  // Remove the groupId and eventSlug keys as we need to create new ones
+  const query = omit(location.query, ['groupSlug', 'eventSlug']);
+  const previousEventUrl = event.previousEventID
+    ? {
+        pathname: location.pathname,
+        query: {
+          ...query,
+          groupSlug: `${event.projectSlug}:${event.groupID}:${event.previousEventID}`,
+        },
+      }
+    : null;
+  const nextEventUrl = event.nextEventID
+    ? {
+        pathname: location.pathname,
+        query: {
+          ...query,
+          groupSlug: `${event.projectSlug}:${event.groupID}:${event.nextEventID}`,
+        },
+      }
+    : null;
+  const newestUrl = {
+    pathname: location.pathname,
+    query: {
+      ...query,
+      groupSlug: `${event.projectSlug}:${event.groupID}:latest`,
+    },
+  };
+  const oldestUrl = {
+    pathname: location.pathname,
+    query: {
+      ...query,
+      groupSlug: `${event.projectSlug}:${event.groupID}:oldest`,
+    },
+  };
+
+  return (
+    <Wrapper>
+      <Container>
+        <StyledLink to={oldestUrl}>
+          <InlineSvg src="icon-prev" size="14px" />
+        </StyledLink>
+        <StyledLink to={previousEventUrl} disabled={previousEventUrl === null}>
+          {t('Older Event')}
+        </StyledLink>
+        <StyledLink to={nextEventUrl} disabled={nextEventUrl === null}>
+          {t('Newer Event')}
+        </StyledLink>
+        <StyledLink to={newestUrl} isLast>
+          <InlineSvg src="icon-next" size="14px" />
+        </StyledLink>
+      </Container>
+    </Wrapper>
+  );
+};
+ModalPagination.propTypes = {
+  location: PropTypes.object.isRequired,
+  event: SentryTypes.Event.isRequired,
+};
+
+const StyledLink = styled(Link, {shouldForwardProp: isPropValid})`
+  color: ${p => (p.disabled ? p.theme.disabled : p.theme.gray3)};
+  font-size: ${p => p.fontSizeMedium};
+  padding: ${space(0.5)} ${space(1.5)};
+  ${p => (p.isLast ? '' : `border-right: 1px solid ${p.theme.borderDark};`)}
+  ${p => (p.disabled ? 'pointer-events: none;' : '')}
+`;
+
+const Wrapper = styled('div')`
+  display: flex;
+`;
+
+const Container = styled('div')`
```

Comment:
We kind of use this and `Wrapper` interchangeably, but having them both here seems awkward. What about `Wrapper` and `InnerWrapper`?

Reply:
How about `ShadowBox` for the inner one?